### ISSUE TITLE
[style.*] and [ngStyle] bindings fail if a trailing semi-colon is present

### ISSUE THREAD

**Issue by cghislai** [angular/angular#6137]
Using angular2 beta0, if one binds to [style] or [ngStyle], the presence of a trailing comma prevent the binding to occur, without any exception thrown to console.

As CSS statements usually end with a semi-colon, this is very confusing.

e.g.:
```
 <div [style.border]="'2px solid yellow;'">I should have a yellow border</div>
  <div [style.border]="'2px solid yellow'">I have a yellow border</div>
```
(http://plnkr.co/edit/c0ZSTpOEvEuo7oMtsBhR?p=preview)

**Comment by pkozlowski-opensource** [comment]
TL;DR; It is a browser that doesn't accept `2px solid yellow;` and fails silently here.

More info: I just did a simple experiment in a browser (Chrome):

```javascript
var div = document.createElement('div');
div.style.border = '2px solid yellow;';
```

Executing this code and printing out content of the `div` variable shows that a desired style is **not** set. So we can conclude that this is behavior **coming from a browser** and not from Angular.

I'm not sure what is the best action here. We could try to truncate a trailing `;` in style expressions (in a renderer) but I suspect that there are other characters that might triggers similar failures in different browsers. We could try to parse those and throw for invalid expressions, but I'm not sure it is worth it.

**Comment by vicb** [comment]
Not an angular bug
